buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/fixtures/TestFixturesPlugin.kt

```diff
@@ -60,46 +55,72 @@ open class TestFixturesPlugin : Plugin<Project> {
         configureAsConsumer()
     }
 
+
+    /**
+     * This mimics what the java-library plugin does, but creating a library of test fixtures instead.
+     */
     private
     fun Project.configureAsProducer() {
+        val testFixtures by java.sourceSets.creating
 
-        configurations {
-            create("outputDirs")
+        java.sourceSets.named<SourceSet>("test") {
```

Comment:
Is there really no pretty way? It feels inconsistent not to use `by getting` here, but then Idea complains about an unused variable.

Reply:
The `SourceSetContainer` is now available as an extension on `Project`. Accessing it that way would shorten this to `sourceSets.named<SourceSet>("test") {`.

The container isn't polymorphic, it should be possible to say `sourceSets.named("test") {`. But it's not, the receiver is then typed `Any`. This is a gap in the kotlin-dsl sugar, thanks! Opened https://github.com/gradle/kotlin-dsl/issues/1104